src/wmtk/MultiMeshManager.cpp

```diff
@@ -0,0 +1,145 @@
+    {
+        auto map_accessor = source_mesh.create_accessor(map_handle);
+        auto map = map_accessor.vector_attribute(source_tuple);
+
+        return std::make_tuple(Tuple(map(0), map(1), map(2), map(3), map(4)), Tuple(map(5), map(6), map(7), map(8), map(9)));
+    }
+
+    void MultiMeshManager::register_child_mesh(Mesh& parent_mesh, std::shared_ptr<Mesh> child_mesh, const std::vector<std::array<Tuple,2>>& child_mesh_simplex_map)
+    {
+        PrimitiveType map_ptype = child_mesh->top_simplex_type();
+        long cur_child_id = long(parent_mesh.multi_mesh_manager.child_meshes.size());
+
+        auto map_to_parent_handle = child_mesh->register_attribute<long>("map_to_parent", map_ptype, 10);
+        auto map_to_child_handle = parent_mesh.register_attribute<long>(fmt::format("map_to_child_{}", cur_child_id), map_ptype, 10);
+        for (long id = 0; id < parent_mesh.capacity(map_ptype); ++id)
+        {
+            write_tuple_map_attribute(map_to_child_handle, parent_mesh, Tuple(), Tuple());
+        }
+
+        // register maps
+        for (long id = 0; id < child_mesh->capacity(map_ptype); ++id)
+        {
+            write_tuple_map_attribute(map_to_parent_handle, *child_mesh, child_mesh_simplex_map[id][0], child_mesh_simplex_map[id][1]);
+
+            write_tuple_map_attribute(map_to_child_handle, parent_mesh, child_mesh_simplex_map[id][1], child_mesh_simplex_map[id][0]);
+        }
+        
+        // update on child_mesh
+        child_mesh->multi_mesh_manager.map_to_parent_handle = map_to_parent_handle;
+        child_mesh->multi_mesh_manager.m_is_parent_mesh = false;
+        child_mesh->multi_mesh_manager.m_child_id = long(parent_mesh.multi_mesh_manager.child_meshes.size());
+
+        // update on parent_mesh
+        parent_mesh.multi_mesh_manager.child_meshes.push_back(child_mesh);
+        parent_mesh.multi_mesh_manager.map_to_child_handles.push_back(map_to_child_handle);
+        parent_mesh.multi_mesh_manager.m_is_parent_mesh = true;
+       
+    }
+
+    Tuple MultiMeshManager::map_tuple_between_meshes(const Mesh& source_mesh, const Mesh& target_mesh, MeshAttributeHandle<long> source_map_handle, const Tuple& source_tuple)
+    {
+        PrimitiveType source_mesh_ptype = source_mesh.top_simplex_type();
+        PrimitiveType target_mesh_ptype = target_mesh.top_simplex_type();
+
+        auto [source_mesh_base_tuple, target_mesh_base_tuple] = read_tuple_map_attribute(source_map_handle, source_mesh, source_tuple);
+
+        if (source_mesh_base_tuple.is_null_tuple())
+        {
+            return Tuple(); // return null tuple
+        }
+        
+        if (source_mesh_ptype == PrimitiveType::Face && target_mesh_ptype == PrimitiveType::Face)
+        {
+            Tuple cur_tuple = source_tuple;
+            std::vector<char> record_switch_operations;
+            while (cur_tuple != source_mesh_base_tuple)
```

Comment:
This should eventually be something like `get_navigation_between_tuples_in_single_simplex(PrimitiveType, const Tuple&, const Tuple&) -> std::vector<PrimitiveType>`. But for now perhaps a function that takes in a `Mesh` + `PrimitiveType` should suffice so you can use mesh navigation to find the sequence.

In the end though, there should be a O(1) solution for computing the navigation (we can precompute all sequences).